src/foam/mlang/mlang.js

```diff
@@ -4807,6 +4836,64 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  package: 'foam.mlang',
+  name: 'If',
+  extends: 'foam.mlang.AbstractExpr',
+
+  properties: [
+    {
+      class: 'foam.mlang.predicate.PredicateProperty',
+      name: 'predicate',
+      javaFactory: 'return foam.mlang.MLang.TRUE;'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'trueExpr'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'falseExpr'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        if ( getTrueExpr() == null ||
```

Comment:
Maybe switch || to &&. It seems legitimate that one would be null.

Reply:
This is only for logging output.   Most of this logging throughout will be removed before any of this is merged.